### DISCUSSION TITLE
Looking for help understanding toolhead coordinate display

### DISCUSSION THREAD

**Discussion by se-SteveThePirate** [mainsail-crew/mainsail#2094] [Help] [answered]
Good afternoon, I'm trying to troubleshoot a bed levelling issue, but before I dig too much further down that rabbit hole, I'd like to make sure I'm correctly understanding what I'm seeing in the Mainsail interface.

TLDR, while troubleshooting bed levelling, I started watching the toolhead coordinates being displayed. For the toolhead purge line printed at the start of the print, the Z coordinate displayed in the large white number in the editable box fluctuates as it prints:
![image](https://github.com/user-attachments/assets/414cf2bd-8048-4763-a5fd-ea7deedcf6eb)
![image](https://github.com/user-attachments/assets/9d122ee2-607b-4f79-ac22-9581d0969cbb)

Then, once it finishes the purge line and starts the print itself, it stays at a static 0.2:
![image](https://github.com/user-attachments/assets/84f91cad-30b7-4e5a-a5c5-a2348e9c54e7)

So my question is, how should I read the small numbers in the grey brackets above the box, versus the larger white numbers inside the box? What's the significance of the number changing during one part of the print, then staying static in the next?

Thanks!

**Comment by meteyou** [comment, marked as answer]
The big numbers are gcode coordinated and the small one are absolute machine coordinates.

If the numbers are not identical, then there are multiple reasons. For example a current movement. The gcode coordinator will be uploaded from target to target point. The machine coordinates are "realtime".
Then there could be a gcode offset (the buttons for z offset for example) or in z also the bed mesh.

**Reply by se-SteveThePirate** [reply]
Thank you for the explanation, I appreciate it.
